index.js

```diff
@@ -0,0 +1,76 @@
+const express = require('express');
+const bcrypt = require('bcryptjs');
+
+const db = require('./data/helpers');
+
+const server = express();
+
+server.use(express.json());
+
+// REGISTER
+server.post('/api/register', async (req, res) => {
+  // get form input values
+  const registrationData = req.body;
+  // validate data
+  if (!registrationData.username || !registrationData.password) {
+    return res
+      .status(400)
+      .json({ message: 'Please enter a valid username and password.' });
+  }
+  // Check if user already exists
+  try {
+    const userInDb = db.getByUsername(registrationData.username);
+    return userInDb
+      ? res.status(422).json({ message: 'That username already exists.' })
+      : null;
```

Comment:
this isn't a great pattern, because it returns no matter what, but only sends a http reply for one fork of the conditional. You should ensure you're returning a HTTP response of one kind or another. I would do this instead:
```
if (userInDb) {
    return res.status(422).json(...etc)
}
```

Reply:
it still probably works in your code because you don't await the promise